File: com/lucciola/haskell/Haskell.java
package com.lucciola.haskell;

import com.lucciola.result.CommunicationResult;

public class Haskell {

	private RexTester compiler;
	private String program;
	private String input;
	
	public Haskell() {
		this.compiler = new RexTester();
		this.program = "";
		this.input = "";
	}
	
	public CommunicationResult run() {
		return this.compiler.post(this.program, this.input);
	}
	
	public void setProgram(String arg0) {
		if (arg0 != null) {
			this.program = arg0;
		} else {
			this.program = "";
		}
	}
	
	public void setInput(String arg0) {
		if (arg0 != null) {
			this.input = arg0;
		} else {
			this.input = "";
		}
	}
	
	public String getProgram() {
		return this.program;
	}
	
	public String getInput() {
		return this.input;
	}
	
}

File: src/main/kotlin/com/lucciola/haskell/Haskell.kt
package com.lucciola.haskell

import com.lucciola.result.CommunicationResult

class Haskell {

    private val compiler: RexTester = RexTester()
    private var program: String = ""
    private var input: String = ""

    fun run(): CommunicationResult {
        // in Kotlin, to generate between non-null type and nullable type, using !! operator
        return this.compiler.post(this.program, this.input)
    }

    fun setProgram(arg0: String?) {
        if (arg0 != null) {
            this.program = arg0
        } else {
            this.program = ""
        }
    }

    fun setInput(arg0: String?) {
        if (arg0 != null) {
            this.input = arg0
        } else {
            this.input = ""
        }
    }

    fun getProgram(): String? {
        return this.program
    }

    fun getInput(): String? {
        return this.input
    }

}
